### ISSUE TITLE
TypeScript compilation issue

### ISSUE THREAD

**Issue by real-artswan** [fox1t/multer#9]
After upgrading
fastify 2.8.0 -> 2.9.0
fastify-multer 1.4.6 -> 1.5.0

I can not compile my project anymore due to the following error:

```
node_modules/fastify-multer/src/lib/content-parser.ts:25:16 - error TS2349: This expression is not callable.
  Type '{ default: <HttpServer = Server, HttpRequest = IncomingMessage, HttpResponse = ServerResponse, T = any>(fn: ((instance: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, options: T, callback: (err?: FastifyError) => void) => void) | { ...; }, options?: string | PluginOptions, next?: nextCallback) => (instance:...' has no call signatures.

25 export default fp(fastifyMulter, {
                  ~~

  node_modules/fastify-multer/src/lib/content-parser.ts:2:1
    2 import * as fp from 'fastify-plugin'
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Type originates at this import. A namespace-style import cannot be called or constructed, and will cause a failure at runtime. Consider using a default import or import require here instead.


Found 1 error.
```

It can be fixed by changing import line in the content-parser.ts to

`import fp from 'fastify-plugin'`

tsc version 3.6.3

**Comment by fox1t** [comment]
Hi @real-artswan thanks for reporting this.
@hakimio is this related with your PR?

**Comment by hakimio** [comment]
@real-artswan do you have `esModuleInterop` set to true in your tsconfig?

**Comment by real-artswan** [comment]
@hakimio yes it is set

**Comment by fox1t** [comment]
This is really strange and I am not understanding what is the problem. I think I have to dig deeper, trying both options.
@real-artswan can you please fix your version to 1.4.6 meanwhile?

**Comment by real-artswan** [comment]
No problem, thanks for your work.

**Comment by atol-kyiv-ua** [comment]
I have the same problem with v1.5.0

**Comment by fox1t** [comment]
I am looking into this now. @artem-tolstykh is `esModuleInterop` set to `true` or `false` in your `tsconfig.json`?

**Comment by atol-kyiv-ua** [comment]
@fox1t it's set.
```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "esModuleInterop": true,
    "resolveJsonModule": true
  },
  "exclude": ["node_modules", "dist"]
}
```

**Comment by fox1t** [comment]
Thanks for sharing. The problem was in fastify-plugin dependency that is not exported to support both default export and commonjs require. I'll reach them to propose a fix. 
However this issue is now fixed in version 1.5.1 already published. Check it out and feel free to open an issue again if still broken.

https://www.npmjs.com/package/fastify-multer
